api/types.proto

```diff
@@ -57,6 +57,13 @@ message GenericResource {
 	}
 }
 
+enum ResourceType {
+	TASK = 0;
+	SECRET = 1;
+	CONFIG = 2;
+	NETWORK = 3;
```

Comment:
`NETWORK` doesn't seem to be used yet. Should we wait until it's supported to add it? @cpuguy83: What did you have in mind here?